src/datanode/src/datanode.rs

```diff
@@ -277,39 +278,45 @@ impl DatanodeBuilder {
         open_with_writable: bool,
     ) -> Result<()> {
         let datanode_table_manager = DatanodeTableManager::new(kv_backend.clone());
+        let table_info_manager = TableInfoManager::new(kv_backend.clone());
         let node_id = self.opts.node_id.context(MissingNodeIdSnafu)?;
-        let mut regions = vec![];
         let mut table_values = datanode_table_manager.tables(node_id);
 
-        while let Some(table_value) = table_values.next().await {
-            let table_value = table_value.context(GetMetadataSnafu)?;
-            for region_number in table_value.regions {
-                regions.push((
-                    RegionId::new(table_value.table_id, region_number),
-                    table_value.engine.clone(),
-                    table_value.region_storage_path.clone(),
-                ));
+        let mut stream = Box::pin(try_stream! {
+            while let Some(table_value) = table_values.try_next().await.context(GetMetadataSnafu)? {
```

Comment:
This will be very slow because it requires looping calls to the `table_info_manager` by table_id.
I think we should add the region `options`  to `TableDataValue` to avoid calling `table_info_manager` in the loop.